src/test/java/LineGraphTest.java

```diff
@@ -52,8 +50,23 @@ public void idempotentReadWrite() throws IOException {
                 diffTrees = LineGraphImport.fromLineGraph(lineGraph, testFile, IMPORT_OPTIONS);
             }
             assertConsistencyForAll(diffTrees);
-            final String lineGraphResult = exportDiffTreeToLineGraph(diffTrees);
-            TestUtils.assertEqualToFile(testFile, lineGraphResult);
+
+            Path actualPath = testFile.getParent().resolve(testFile.getFileName().toString() + ".actual");
+            try (var output = IO.newBufferedOutputStream(actualPath)) {
+                LineGraphExport.toLineGraphFormat(diffTrees, EXPORT_OPTIONS, output);
+            }
+
+            try (
+                    var expectedFile = Files.newBufferedReader(testFile);
+                    var actualFile = Files.newBufferedReader(actualPath);
+            ) {
+                if (!IOUtils.contentEqualsIgnoreEOL(expectedFile, actualFile)) {
+                    Assert.fail("The file " + testFile + " couldn't be exported or imported without modifications");
+                } else {
+                    // Only keep output file on errors
+                    Files.delete(actualPath);
+                }
+            }
```

Comment:
Looks good. Previously, this test case never wrote to disk though and now we have several extra file system interactions. But I guess that is fine and we should not complicate the test to avoid that.

Reply:
I didn't do this because its simpler (it's actually pretty equally simple to keep the line graph in memory using `ByteArrayOutputStream`), but because it has the following two benefits:
- Reduces memory usage: I care at least a little bit about the memory footprint (in this case at least 128K) and let the OS decide whether its worth the effort to write that file on disk.
- More important: It simplifies testing by letting me diff the resulting files if a test case fails.
- There is a third reason for this change which doesn't apply here, but to the same pattern in #54: If I change the code in a way that the test case has to be modified I can easily replace the expected output file by the actual output file after careful inspection.

Reply:
ok